catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -190,4 +248,26 @@ private void copyQueryRequestAndQuery(QueryRequest queryRequest) {
                     queryRequest.getProperties());
         }
     }
+
+    private static class ResultListIterator implements Iterator<List<Result>> {
```

Comment:
✏️  If we keep this here, should we keep the iterator inner-class names consistent, i.e., name this one `QueryResultListIterator` or name the other one `ResultIterator`?